### DISCUSSION TITLE
Copilot will make requets via host proxies on server

### DISCUSSION THREAD

**Discussion by scruel** [community/community#74788] [Copilot Conversations]
### Select Topic Area

Feature request

### Body

For remote developing (on windows host, using SSH to connect the Ubuntu server), copilot will make requests on the server with the host proxies, which caused connection issues.

`settings.json` on host:
```
{
  ...
  "http.proxy": "socks5://corp.example.com:8080"
}
```

Using SSH remotely connection to the server, leave `settings.json` on server empty, plugin will keep producing errors:
```
Proxy socket connection error,Failed to establish a socket connection to proxies: ["PROXY corp.example.com:8080"]
```

After a long time of trying without success, I had to set up proxies on our server, but it will produce another [error](https://github.com/orgs/community/discussions/74997).

Now if we remove proxies from settings, we can get rid of the host proxies settings, however, we won't be able to connect:
```
## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- default.exp-tas.com: connect ETIMEDOUT 13.107.5.93:443

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy autentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

```

BTW, I have proxies on my host rather than on every server, can you provide an option to let copilot make requests via the host proxies on our host while coding remotely via SSH, instead of on the servers?

**Comment by gitstua** [comment]
Hi @scruel 

There are a couple of things to consider reading:
- [Troubleshooting GitHub Copilot](https://docs.github.com/en/enterprise-cloud@latest/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot)
- Ensure your firewall/proxy team are allowing all the URLs listed on the [troubleshooting firewall](https://docs.github.com/en/enterprise-cloud@latest/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot) page

**Reply by scruel** [reply]
Those articles are not helpful… I updated the description, can you read it again?
